lib/repl.js

```diff
@@ -257,6 +257,14 @@ function REPLServer(prompt,
         }
       } catch (e) {
         err = e;
+
+        if (err === null || err === undefined) {
+          debug('not recoverable, send to domain, throw null or undefined');
+          process.domain.emit('error', err);
+          process.domain.exit();
+          return;
+        }
```

Comment:
You can merge this with line 266-269 if you change the condition on line 260 to `err && err.message === ...` and line 275 to just `if (process.domain) {`.

Can you also add a regression test?  Thanks.

Reply:
I wannted to add test but the thing is I Dont know how to! I checkout guide on writing the test.

Reply:
`test/parallel/test-repl-domain.js` is probably the right place for it, although if you can fit it in one of the other `test/parallel/test-repl-*` tests, that's probably perfectly alright too.

Reply:
@bnoordhuis added regression test and fixed changes requested.
But i dont know about line 275 `if (process.domain) {`.